metrics-proxy/src/main/java/ai/vespa/metricsproxy/http/application/ApplicationMetricsHandler.java

```diff
@@ -64,4 +79,16 @@ private JsonResponse applicationMetricsResponse(String requestedConsumer) {
         }
     }
 
+    private TextResponse applicationPrometheusResponse(String requestedConsumer) {
+        ConsumerId consumer = getConsumerOrDefault(requestedConsumer, metricsConsumers);
+        var metricsByNode = metricsRetriever.getMetrics(consumer);
+
+
+        return new TextResponse(200, PrometheusUtil.toPrometheusModel(toGenericApplicationModel(metricsByNode).nodes.stream()
```

Comment:
This section was a bit difficult to read. I experimented a bit, and came up with this. Perhaps you can come up with something even better:
```
        List<GenericJsonModel> genericNodes = toGenericApplicationModel(metricsByNode).nodes;
        List<MetricsPacket> metricsForAllNodes = genericNodes.stream()
                .flatMap(element -> toMetricsPackets(element).stream()
                        .map(MetricsPacket.Builder::build))
                .collect(Collectors.toList());
        return new TextResponse(200, toPrometheusModel(metricsForAllNodes).serialize());
```